qa/rpc-tests/test_framework/util.py

```diff
@@ -318,27 +318,51 @@ def set_node_times(nodes, t):
     for node in nodes:
         node.setmocktime(t)
 
+def wait_until(predicate, attempts=float('inf'), timeout=float('inf')):
+    attempt = 0
+    elapsed = 0
+
+    while attempt < attempts and elapsed < timeout:
+        if predicate():
+            return True
+        attempt += 1
+        elapsed += 0.1
+        time.sleep(0.1)
+
+    return False
+
```

Comment:
Given this utility function can be reused in the future and the `predicate()` evaluation can potentially take some time to execute, why not refactoring the time control (`timeout` and `elapsed`) so that it actually checks for executing time?
Something like:

```suggestion
def wait_until(predicate, attempts=float('inf'), timeout=float('inf')):
    attempt = 0
    elapsed = time.time()

    while attempt < attempts and (time.time() - elapsed) < timeout:
        if predicate():
            return True
        attempt += 1
        time.sleep(0.1)

    return False
```

Reply:
Yes, it sounds reasonable and I like this improvement